File: uniffi_bindgen/src/bindings/kotlin/templates/CallbackInterfaceTemplate.kt
{%- let cbi = ci|get_callback_interface_definition(name) %}
{%- let callback_handler_class = format!("UniffiCallbackInterface{}", name) %}
{%- let callback_handler_obj = format!("uniffiCallbackInterface{}", name) %}
{%- let ffi_init_callback = cbi.ffi_init_callback() %}
{%- let interface_name = cbi|type_name(ci) %}
{%- let methods = cbi.methods() %}
{%- let interface_docstring = cbi.docstring() %}

{% include "Interface.kt" %}
{% include "CallbackInterfaceImpl.kt" %}

// The ffiConverter which transforms the Callbacks in to Handles to pass to Rust.
public object {{ ffi_converter_name }}: FfiConverterCallbackInterface<{{ interface_name }}>()

File: uniffi_bindgen/src/bindings/python/templates/CallbackInterfaceTemplate.py
{%- let protocol = cbi.protocol %}
{%- let vtable = cbi.vtable %}
{%- let trait_impl=format!("_UniffiTraitImpl{}", cbi.name) %}
{%- let ffi_converter_name = cbi.self_type.ffi_converter_name %}

{% include "Protocol.py" %}
{% include "CallbackInterfaceImpl.py" %}

# The _UniffiConverter which transforms the Callbacks in to Handles to pass to Rust.
{{ ffi_converter_name }} = _UniffiCallbackInterfaceFfiConverter()
